go/vt/tabletmanager/state_change.go

```diff
@@ -293,6 +293,27 @@ func (agent *ActionAgent) changeCallback(ctx context.Context, oldTablet, newTabl
 		}
 	}
 
+	// See if we need to enable or disable semi-sync replication.
+	// Note we never enable the master side, as that would have
+	// been done by either the reparent process, or an external
+	// tool (like Orchestrator).
+	if *enableSemiSync {
```

Comment:
I don't think so, ActionAgent.enableSemiSync is for the new state only, this is for a transition.